core/src/impl/Kokkos_FunctorAdapter.hpp

```diff
@@ -115,21 +127,27 @@ struct ReduceFunctorHasFinal {
 };
 
 #if defined(KOKKOS_COMPILER_MSVC) || defined(KOKKOS_IMPL_WINDOWS_CUDA)
-template <class F>
-using final_archetype = decltype(&F::final);
-
 template <class FunctorType>
 struct ReduceFunctorHasFinal<
-    FunctorType, impl_void_t_workaround<final_archetype<FunctorType>>> {
+    FunctorType, impl_void_t_workaround<decltype(&FunctorType::final)>> {
   enum : bool { value = true };
 };
 #else
+// FIXME_SYCL not all compiler recognize this as separate specialization
+#ifdef KOKKOS_ENABLE_SYCL
```

Comment:
Did you mean to have it here and not after the first specialization of `ReduceFunctorHasFinal` below?

Reply:
Let me just improve the comment here to clarify which specialization I mean.

Reply:
Ping